src/Sylius/Bundle/ApiBundle/Serializer/ProductImageNormalizer.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use Sylius\Component\Core\Model\ProductImageInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+class ProductImageNormalizer implements ContextAwareNormalizerInterface, NormalizerAwareInterface
+{
+    use NormalizerAwareTrait;
+
+    private const ALREADY_CALLED = 'product_image_normalizer_already_called';
+
+    /** @var string */
+    private $prefix;
+
+    public function __construct(string $prefix)
+    {
+        $this->prefix = $this->validatePrefix($prefix);
+    }
+
+    public function normalize($object, $format = null, array $context = [])
+    {
+        Assert::isInstanceOf($object, ProductImageInterface::class);
+        Assert::keyNotExists($context, self::ALREADY_CALLED);
+
+        $context[self::ALREADY_CALLED] = true;
+
+        $data = $this->normalizer->normalize($object, $format, $context);
+
+        $data['media_path'] = $this->prefix . $data['path'];
```

Comment:
Is `media_path` added to the endpoint documentation? Nonetheless, I would remove this additional field and just replace the value of `path`. Without prefix it is useless.
```suggestion
        $data['path'] = $this->prefix . $data['path'];
```